tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/RetryClientTest.java

```diff
@@ -0,0 +1,154 @@
+        CountDownLatch countDownLatchForRestart = new CountDownLatch(1);
+        client.setCountDownLatch(countDownLatchForRestart);
+        remoteServer.run();
+        countDownLatchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        client.sendBinary(bufferData);
+        countDownLatchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getBufferReceived(), bufferData);
+        client.shutDown();
+        remoteServer.stop();
+    }
+
+    @Test(description = "Tests the retry function using the WebSocket client (starting the server " +
+            ", sending, and receiving binary frames then restart that server and repeat.)")
+    public void testBinaryFrameForRetryWithMaxCount() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        ByteBuffer bufferSent = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5});
+        remoteServer = new WebSocketRemoteServer(port);
+        remoteServer.run();
+        WebSocketTestClient client = new WebSocketTestClient("ws://localhost:21031");
+        client.handshake();
+        client.setCountDownLatch(countDownLatch);
+        client.sendBinary(bufferSent);
+        countDownLatch.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getBufferReceived(), bufferSent);
+        remoteServer.stop();
+        CountDownLatch latchForRestart = new CountDownLatch(1);
+        client.setCountDownLatch(latchForRestart);
+        latchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        remoteServer.run();
+        client.sendBinary(bufferSent);
+        latchForRestart.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertNull(client.getBufferReceived());
+        client.shutDown();
+        remoteServer.stop();
+    }
+
+    @Test(description = "Tests the `countDownLatch` for the retry function using the WebSocket client (starting the" +
+            " server, sending, and receiving binary frames then restart that server and wait for six minutes.)")
+    public void testCoundownLatchForRetry() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        ByteBuffer bufferSent = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5});
+        remoteServer = new WebSocketRemoteServer(port);
+        remoteServer.run();
+        WebSocketTestClient client = new WebSocketTestClient(url);
+        client.handshake();
+        client.setCountDownLatch(countDownLatch);
+        client.sendBinary(bufferSent);
+        countDownLatch.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getBufferReceived(), bufferSent);
+        remoteServer.stop();
+        CountDownLatch latchForRestart = new CountDownLatch(1);
+        client.setCountDownLatch(latchForRestart);
+        latchForRestart.await(320, TimeUnit.SECONDS);
```

Comment:
Are you really going to wait for 6 minutes in an integration test 😮

Reply:
We wait 5 minutes and 20 sec because As per the implementation https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/InitEndpoint.java#L102, we have to wait 5 minutes for check whether the counDownLatch functionality works or not

Reply:
I remember asking you to make that parameter configurable so that it need not wait for 5 minutes